src/NzbDrone.Core/Datastore/ConnectionStringFactory.cs

```diff
@@ -74,5 +90,61 @@ private DatabaseConnectionInfo GetPostgresConnectionString(string dbName)
 
             return new DatabaseConnectionInfo(DatabaseType.PostgreSQL, connectionBuilder.ConnectionString);
         }
+
+        private DatabaseConnectionInfo GetPostgresConnectionInfoFromConnectionString(string connectionString)
+        {
+            var connectionBuilder = new NpgsqlConnectionStringBuilder(connectionString)
+            {
+                Enlist = false
+            };
+
+            return new DatabaseConnectionInfo(DatabaseType.PostgreSQL, connectionBuilder.ConnectionString);
+        }
+
+        /// <summary>
+        /// Validates that either Postgres connection strings are both set or neither are set, and that either connection strings or
+        /// other Postgres settings are set, but not both.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when configuration is invalid.</exception>
+        private void ValidatePostgresOptions()
+        {
+            var isMainDBConnectionStringSet = !string.IsNullOrWhiteSpace(_configFileProvider.PostgresMainDbConnectionString);
+            var isLogDBConnectionStringSet = !string.IsNullOrWhiteSpace(_configFileProvider.PostgresLogDbConnectionString);
+            var isHostSet = !string.IsNullOrWhiteSpace(_configFileProvider.PostgresHost);
```

Comment:
Use the extension for not null/whitespace.

```suggestion
            var isMainDBConnectionStringSet = _configFileProvider.PostgresMainDbConnectionString.IsNotNullOrWhiteSpace(;
            var isLogDBConnectionStringSet = _configFileProvider.PostgresLogDbConnectionString.IsNotNullOrWhiteSpace(;
            var isHostSet = _configFileProvider.PostgresHost.IsNotNullOrWhiteSpace();
```